### ISSUE TITLE
`from_dlpack` causes segmentation fault on unsupported dtype

### ISSUE THREAD

**Issue by toslunar** [cupy/cupy#8801]
### Description

Converting torch.Tensor of dtype `bfloat16` caused segmentation fault. I'd get Python exception for it to handle the error.

DLPack has added bfloat16 support: https://github.com/dmlc/dlpack/issues/45.
Just FYI, PyTorch raises TypeError for NumPy not supporting bfloat16.
```
>>> torch.arange(3, dtype=torch.bfloat16).numpy()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Got unsupported ScalarType BFloat16
```

### To Reproduce

```py
import cupy, torch
cupy.from_dlpack(torch.arange(3, dtype=torch.bfloat16, device="cuda:0"))
```

### Installation

Wheel (`pip install cupy-***`)

### Environment

```
OS                           : Linux-6.8.0-44-generic-x86_64-with-glibc2.35
Python Version               : 3.12.7
CuPy Version                 : 13.3.0
CuPy Platform                : NVIDIA CUDA
NumPy Version                : 2.0.2
SciPy Version                : 1.14.1
Cython Build Version         : 0.29.36
Cython Runtime Version       : None
CUDA Root                    : /usr/local/cuda
nvcc PATH                    : /usr/local/cuda/bin/nvcc
CUDA Build Version           : 12060
CUDA Driver Version          : 12060
CUDA Runtime Version         : 12060 (linked to CuPy) / 12010 (locally installed)
CUDA Extra Include Dirs      : []
cuBLAS Version               : 120103
cuFFT Version                : 11002
cuRAND Version               : 10302
cuSOLVER Version             : (11, 4, 5)
cuSPARSE Version             : 12100
NVRTC Version                : (12, 1)
Thrust Version               : 200600
CUB Build Version            : 200600
Jitify Build Version         : <unknown>
cuDNN Build Version          : 8801
cuDNN Version                : 8900
NCCL Build Version           : 21602
NCCL Runtime Version         : 21701
cuTENSOR Version             : None
cuSPARSELt Build Version     : None
Device 0 Name                : Tesla V100-SXM2-32GB
Device 0 Compute Capability  : 70
Device 0 PCI Bus ID          : 0000:1A:00.0
Device 1 Name                : Tesla V100-SXM2-32GB
Device 1 Compute Capability  : 70
Device 1 PCI Bus ID          : 0000:1B:00.0
Device 2 Name                : Tesla V100-SXM2-32GB
Device 2 Compute Capability  : 70
Device 2 PCI Bus ID          : 0000:3D:00.0
Device 3 Name                : Tesla V100-SXM2-32GB
Device 3 Compute Capability  : 70
Device 3 PCI Bus ID          : 0000:3E:00.0
Device 4 Name                : Tesla V100-SXM2-32GB
Device 4 Compute Capability  : 70
Device 4 PCI Bus ID          : 0000:88:00.0
Device 5 Name                : Tesla V100-SXM2-32GB
Device 5 Compute Capability  : 70
Device 5 PCI Bus ID          : 0000:89:00.0
Device 6 Name                : Tesla V100-SXM2-32GB
Device 6 Compute Capability  : 70
Device 6 PCI Bus ID          : 0000:B2:00.0
Device 7 Name                : Tesla V100-SXM2-32GB
Device 7 Compute Capability  : 70
Device 7 PCI Bus ID          : 0000:B3:00.0
```

**Comment by kmaehashi** [comment]
Thanks! Confirmed on our end.

Worth noting this issue does not happen with CuPy v14, which has overhauled DLPack implementation that supports DLPack v1 #8683 

```
>>> cupy.from_dlpack(torch.arange(3, dtype=torch.bfloat16, device="cuda:0"))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "cupy/_core/dlpack.pyx", line 566, in cupy._core.dlpack.from_dlpack
  File "cupy/_core/dlpack.pyx", line 447, in cupy._core.dlpack._dlpack_to_cupy_array
NotImplementedError: CuPy does not support bfloat16 yet
```

**Comment by seberg** [comment]
You backported those changes already.  So they should already be included as soon as a 13.4.0 release happens.

**Comment by kmaehashi** [comment]
Ah, you're right! It should be fixed in the upcoming release. Thanks again @toslunar for reporting! 🙏🏼
